timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -61,6 +62,23 @@ void verify_coords_and_box(
     }
 }
 
+// accumulate_energies handles the combining of energies, summing them up deterministically
+// and returning nan if there are overflows.
+// The energies are collected in unsigned long long and overflows are incremented or decremented
+// by the potentials. Currently only the negated NonbondedAllPairs decrements the overflows which
+// are expected to exactly cancel out with other Nonbonded potentials if there are no clashes.
+double accumulate_energies(int N, unsigned long long *fixed_u, int overflows) {
```

Comment:
Mostly for simplicity. If we simply return a single value (+inf/nan), the exclusions can decrement and the nonbonded can increment. Otherwise, they would need to both increment/decrement according to the sign of the overflow, which is more difficult to debug. Though the `+inf/-inf` distinction would be useful to the end user? 

Is there a use case for distinguishing between for +/- inf?